[BUG] test_claude_worker_server.py: 11 testes defasados vs contrato de dispatch/lease (lease_path + claude_pid/claude_running) — suíte completa vermelha

## Descrição do Bug

A suíte completa (`python3 -m pytest deile/tests/ -q`) falha com **11 testes vermelhos em `deile/tests/infrastructure/test_claude_worker_server.py`** que falham de forma **determinística** (também em isolamento, `-p no:randomly -p no:cov`). Não é poluição de ordenação — é **divergência teste↔código**: o `claude_worker_server.py` evoluiu o contrato do `/v1/dispatch` e do lease, mas os mocks/asserts do teste não foram atualizados junto.

Esses 11 vermelhos, somados a outras ~13 falhas de ordering-pollution (rastreadas à parte), mantêm a **suíte completa permanentemente vermelha**, o que **trava o quality-gate de "suíte completa verde"** do pipeline — toda PR bot-autorada esgota o teto de tentativas no reaper sem mergear (vide #490 e #489, que precisaram de merge manual).

## Severidade

- [x] **Maior** — uma funcionalidade central está quebrada ou produz resultado errado; sem contorno

> O auto-merge do pipeline depende da suíte completa verde; com a suíte vermelha, nenhuma PR bot-autorada mergeia sozinha. Contorno atual = merge manual pelo Humano.

## Passos para Reproduzir

1. A partir da raiz do repo, na `main` (commit `54fe614` ou posterior):
2. `python3 -m pytest deile/tests/infrastructure/test_claude_worker_server.py -q -p no:randomly -p no:cov`
3. Observar **11 failed, 73 passed**.

## Comportamento Esperado

`test_claude_worker_server.py` passa 100% (84 passed), refletindo o contrato atual de `/v1/dispatch` e do lease.

## Comportamento Atual

11 falhas determinísticas, em dois grupos com a **mesma causa-raiz** (commit `b5f3672`, PR #467):

**Grupo A — 10 testes de dispatch retornam HTTP 500** (cascata de um único `TypeError`):
- `test_dispatch_translates_model_slug`, `test_dispatch_response_shape`, `test_dispatch_creates_workspace_dir`, `test_dispatch_returns_auth_expired_when_claude_reports_not_logged_in`, `test_dispatch_returns_ok_when_claude_succeeds_normally`, `test_dispatch_persists_session_metadata`, `test_dispatch_resume_uses_minus_r_flag`, `test_dispatch_passes_brief_with_preamble` (IndexError), `test_dispatch_passes_session_id_flag_to_claude` (KeyError: 'args'), `test_dispatch_detects_auth_expired_via_json_output` (KeyError: 'error_code').
- Causa: o handler de dispatch passou a invocar o runner com o kwarg **`lease_path=`**, mas os fakes do teste (`fake_run`/mocks de subprocess) não aceitam esse argumento → `TypeError` → o handler captura e devolve **500** → os asserts seguintes (`error_code`, `args`, shape) caem em cascata.

**Grupo B — 1 teste de redação do lease:**
- `test_pod_status_endpoint_redacts_lease_payload` — `AssertionError: Lease exposed extra fields: {'claude_pid', 'claude_running'}`.
- Causa: o lease ganhou os campos **`claude_pid`** e **`claude_running`** (liveness real do subprocess `claude -p`), mas o teste ainda espera que o payload do endpoint de status NÃO contenha campos extras.

## Saída de Erro

```
ERROR deile.claude_worker_server:claude_worker_server.py:2298 dispatch failed task_id=...
Traceback (most recent call last):
  File ".../infra/k8s/claude_worker_server.py", line 2293, in dispatch_handler
TypeError: test_dispatch_response_shape.<locals>.fake_run() got an unexpected keyword argument 'lease_path'

=== short test summary ===
FAILED ...::test_dispatch_translates_model_slug - AssertionError: assert 500 == 200
FAILED ...::test_dispatch_response_shape - AssertionError: assert 500 == 200
FAILED ...::test_dispatch_creates_workspace_dir - AssertionError: assert 500 == 200
FAILED ...::test_dispatch_passes_brief_with_preamble - IndexError: list index out of range
FAILED ...::test_dispatch_returns_auth_expired_when_claude_reports_not_logged_in - assert 500 == 200
FAILED ...::test_dispatch_returns_ok_when_claude_succeeds_normally - assert False is True
FAILED ...::test_dispatch_persists_session_metadata - assert True is False
FAILED ...::test_dispatch_passes_session_id_flag_to_claude - KeyError: 'args'
FAILED ...::test_dispatch_resume_uses_minus_r_flag - AssertionError: assert 500 == 200
FAILED ...::test_dispatch_detects_auth_expired_via_json_output - KeyError: 'error_code'
FAILED ...::test_pod_status_endpoint_redacts_lease_payload - Lease exposed extra fields: {'claude_pid', 'claude_running'}
11 failed, 73 passed
```

## Causa-raiz (commit/PR identificados)

- **Commit `b5f3672`** — `chore(observability): mistery fixes — procps, claude_pid no lease, branch órfã`
- **Mergeado pela PR #467** (`chore/observability-mistery-fixes`, 31/mai/2026, merge `a57ddb1`)
- Esse commit introduziu **simultaneamente** o kwarg `lease_path=` na chamada do runner **e** os campos `claude_pid`/`claude_running` no lease — sem atualizar `test_claude_worker_server.py`.

**Prova de que NÃO é regressão do #445/#497** (que mexeu no mesmo arquivo): trocando apenas o `claude_worker_server.py` pela versão pré-#497 (`668ee7e~1`) e rerodando o teste, o resultado é **idêntico (11 failed, 73 passed)** — as falhas precedem o #497.

## Correção proposta (atualizar os TESTES, o código está correto)

1. **Grupo A:** atualizar os fakes/mocks de subprocess em `test_claude_worker_server.py` para **aceitar o kwarg `lease_path=`** (e quaisquer kwargs novos via `**kwargs`). Após isso, os asserts de shape/`error_code`/`args` devem voltar a passar (a cascata some quando o 500 some).
2. **Grupo B:** atualizar `test_pod_status_endpoint_redacts_lease_payload` para **aceitar `claude_pid` e `claude_running`** como campos legítimos do lease. Confirmar (threat model §7) que são não-sensíveis — `claude_pid` é um PID, `claude_running` é booleano; nenhum é segredo, então expô-los no endpoint de status é intencional (são a fonte de verdade de liveness, vide CLAUDE.md). A redação deve continuar removendo apenas env/segredos.
3. **Gate de aceite:** `python3 -m pytest deile/tests/infrastructure/test_claude_worker_server.py -q -p no:randomly -p no:cov` → **84 passed, 0 failed**. Rodar também a suíte completa e confirmar que o número de falhas cai de 24 para ~13 (as restantes são a ordering-pollution rastreada à parte).

## Escopo (NÃO fazer aqui)

- **NÃO** tocar nas ~13 falhas de ordering-pollution (`test_dispatch_resolver_settings.py`, `test_deile_md_loader.py`) — são bug separado (regressão da #432), rastreado em issue própria.
- **NÃO** alterar o comportamento de runtime do `claude_worker_server.py` — o contrato atual (`lease_path`, `claude_pid`, `claude_running`) é o desejado; só os testes estão defasados.

## Ambiente

- Repo `elimarcavalli/deile`, branch `main` (commit `54fe614`).
- Python 3.9+ (reproduzido em 3.14 local e na imagem `deile-stack:local`).
- `pytest` com `pytest-randomly`; reproduz com e sem randomização.

--- a/deile/tests/infrastructure/test_claude_worker_server.py
+++ b/deile/tests/infrastructure/test_claude_worker_server.py
@@ -225,7 +225,7 @@ async def test_dispatch_translates_model_slug(
     """
     captured = {}
 
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         captured["args"] = list(args)
         return claude_worker_module.SubprocessResult(
             returncode=0, stdout="ok\n", stderr="", duration_seconds=1.0,
@@ -266,7 +266,7 @@ async def test_dispatch_response_shape(
     ``session_id``, ``attempt``, ``duration_seconds`` e ``returncode``
     — contrato consumido pelo ``deile-pipeline`` e pelo painel TUI."""
     import json as _json
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         # Com --output-format json, stdout é UM JSON object (resultado final).
         out = _json.dumps({
             "is_error": False, "result": "ok", "session_id": "abc-123",
@@ -314,7 +314,7 @@ async def test_dispatch_creates_workspace_dir(
     arquivos/staged changes."""
     captured_cwd = []
 
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         captured_cwd.append(cwd)
         assert cwd.exists(), f"workspace {cwd} should exist before exec"
         return claude_worker_module.SubprocessResult(0, "", "", 0.1)
@@ -348,7 +348,7 @@ async def test_dispatch_passes_brief_with_preamble(
     ``$BRANCH`` é substituído no template antes do exec."""
     captured_args = []
 
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         captured_args.extend(args)
         return claude_worker_module.SubprocessResult(0, "", "", 0.1)
 
@@ -522,7 +522,7 @@ async def test_dispatch_returns_auth_expired_when_claude_reports_not_logged_in(
     from infra.k8s import claude_worker_server as mod
     import types
 
-    async def fake_run_subprocess(args, *, cwd, task_id, timeout):
+    async def fake_run_subprocess(args, *, cwd, task_id, timeout, lease_path=None):
         return types.SimpleNamespace(
             returncode=1,
             stdout="Not logged in · Please run /login\n",
@@ -558,7 +558,7 @@ async def test_dispatch_returns_ok_when_claude_succeeds_normally(
     import json as _json
     import types
 
-    async def fake_run_subprocess(args, *, cwd, task_id, timeout):
+    async def fake_run_subprocess(args, *, cwd, task_id, timeout, lease_path=None):
         out = _json.dumps({
             "is_error": False,
             "result": "Review completed STATUS: APPROVE",
@@ -601,7 +601,7 @@ async def test_dispatch_persists_session_metadata(
     session_id, workdir, stage, attempt=1, started_at, last_*."""
     import json as _json
 
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         out = _json.dumps({
             "is_error": False, "result": "done", "session_id": "fake-sess",
             "total_cost_usd": 0.07, "duration_ms": 5000, "num_turns": 2,
@@ -644,7 +644,7 @@ async def test_dispatch_passes_session_id_flag_to_claude(
     import json as _json
     captured = {}
 
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         captured["args"] = list(args)
         out = _json.dumps({"is_error": False, "result": "ok", "session_id": "x"})
         return claude_worker_module.SubprocessResult(0, out, "", 1.0)
@@ -682,7 +682,7 @@ async def test_dispatch_resume_uses_minus_r_flag(
     import json as _json
     captured = {}
 
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         captured["args"] = list(args)
         captured["cwd"] = cwd
         captured["task_id"] = task_id
@@ -846,7 +846,7 @@ async def test_dispatch_detects_auth_expired_via_json_output(
     'Not logged in' → ok=False, error_code=WORKER_AUTH_EXPIRED.
     """
     import json as _json
-    async def fake_run(args, *, cwd, task_id, timeout):
+    async def fake_run(args, *, cwd, task_id, timeout, lease_path=None):
         out = _json.dumps({
             "is_error": True,
             "result": "Not logged in · Please run /login",
@@ -1808,7 +1808,7 @@ async def test_pod_status_endpoint_redacts_lease_payload(
         body = await resp.json()
     lease = body["lease"]
     assert lease is not None
-    allowed_keys = {"task_id", "heartbeat_at", "pid"}
+    allowed_keys = {"task_id", "heartbeat_at", "pid", "claude_pid", "claude_running"}
     assert set(lease.keys()) <= allowed_keys, (
         f"Lease exposed extra fields: {set(lease.keys()) - allowed_keys}"
     )

